### PR TITLE
feat: adds logo component and improves the header/footer components

### PR DESCRIPTION
**Pull request by joetower**
**Summary**

This PR fixes/implements the following **bugs/features**

- Adds new logo component
- Pulls the logo component into both `site--header` and `site--footer` components.
- Wraps the logo in a link component (`link_content`)
- I view this PR as an incremental improvement to the `header` component as the `main-menu` does not yet render. Once the `main-menu` is worked on the header component _should_ pull it in properly.

Explain the **motivation** for making this change. What existing problem does the pull request solve?

**Documentation Update (required)**

If this pull request requires a change to Emulsify documentation, those changes, updates, and/or new information must accompany this pull request.

**How to review this PR**

- [x] Checkout branch and run `npm run develop`
- [x] Navigate to http://localhost:6006/?path=/story/molecules-logo--logo and verify you see a logo component render
- [x] Navigate to http://localhost:6006/?path=/story/organisms-site--header and http://localhost:6006/?path=/story/organisms-site--footer and verify the logo renders on each with a `header`/`footer` modifiers.
- [x] Also view the site templates and see both header/footer instances together http://localhost:6006/?path=/story/templates-layouts--full-width

### CODE REVIEW

**Review by Gregg-Hendrix — APPROVED** [review]
Hey Joe! I reviewed these changes, and they look good to me. I navigated to all of the links that you posted, and saw that the logo does appear in the `header` and `footer`. They also appear in the full-width component! I can verify that everything looks good, but perhaps someone else should review the actual code? Just in case there are improvements that I don't yet have the knowledge to give.  I've taken a look at your code as well, and it looks spot on to me!

**Review by callinmullaney — CHANGES_REQUESTED** [review]
@joetower This PR is looking pretty good but I think it needs some edits before we should review as a group:
1. In the logo.stories.js file we should expand upon line 8 so instead of just `logoExample` it should be `headerLogo` and pass the appropriate modifier. Then underneath that we have a `footerLogo` and similarly passed the modifier. If you think this is too verbose then I would just change `logoExample` to `logo` to keep storybook consistent.
2. `logo.twig` is missing Available Variables documentation at the top of the file. See other components/PRs for example on how to structure this comment.

File: components/02-molecules/logo/logo.twig [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{% set logo__base_class = logo__base_class|default('logo') %}
+
+{% set logo__attributes = {
+  class: bem(logo__base_class, logo__modifiers),
```
Comment (callinmullaney) [inline components/02-molecules/logo/logo.twig:4]: missing `logo__blockname`

**Comment by joetower** [comment]
@callinmullaney Thank you for the feedback! This should be ready for another look.

**Review by ModulesUnraveled — CHANGES_REQUESTED** [review]
Simplify per FEPG review

**Comment by joetower** [comment]
@ModulesUnraveled Think I addressed all of the changes we discussed in the PG. Please take a look and let me know if I missed anything. Thank you!

**Comment by codeb-a** [comment]
Hey Joe. I can verify that the the logo is rendering for the header and the footer and also for the page layout. but nothing is rendering here when I click the link  http://localhost:6006/?path=/story/molecules-logo--logo-example. Otherwise its functioning as intended.

**Comment by joetower** [comment]
> Hey Joe. I can verify that the the logo is rendering for the header and the footer and also for the page layout. but nothing is rendering here when I click the link http://localhost:6006/?path=/story/molecules-logo--logo-example. Otherwise its functioning as intended.

@codeb-a Thank you! This was due to not updating the testing instructions after re-naming the component. Sorry about that. This is the correct component link: http://localhost:6006/?path=/story/molecules-logo--logo and I updated the link in the PR testing instructions.

**Comment by joetower** [comment]
@callinmullaney This should be ready for another look when you have time. I realized Brian never reviewed it again after I made updates.

**Review by mcortes19 — APPROVED** [review]

**Review by mariannuar — APPROVED** [review]

**Pull request merged**